nengo/cache.py

```diff
@@ -411,10 +415,9 @@ def _write_index(self):
                 "software. It is safe to ignore this warning. But if you see "
                 "it a lot, you might want to consider doing one of the "
                 "following for the best Nengo performance:\n"
-                "1. Try using Python 3 instead of Python 2.\n"
-                "2. Configure your anti-virus to ignore the Nengo cache "
+                "1. Configure your anti-virus to ignore the Nengo cache "
                 "folder ('{cache_dir}').\n"
-                "3. Disable the cache.\n"
+                "2. Disable the cache.\n"
```

Comment:
Could this happen if the user simply doesn't have permission to write to the cache dir? Should we add that as one of the steps to check, before disabling the cache?

Reply:
I don't think we really have a good grasp on why this happens or how to fix it.  For example in this case it was failing non-deterministically, so it's not a simple permission error.  Until we know we have a fix that works, I'm inclined to just leave it as the general "ignore this warning, or disable the cache".  I even thought about taking out the anti-virus part (since we don't really know if that's causing the issue either), but left it in as at least we have one well-documented case where that seemed to fix it.

Reply:
If you have the cache in a folder that you explicitly do not have permissions for, do you get a permission error consistently? If that's the case, then I would suggest checking permissions.

As for the anti-virus part, I'm not sure who put that in or how well supported that is. I had assumed it was more related to the CalledProcessError, but who knows.

Reply:
I did a quick test, and it looks like (at least on my system) if you don't have permissions for the cache dir, you get an error in `DecoderCache.__enter__`, and the warning there gets displayed. It's not terribly helpful, because we group all errors together and display the same generic message for all of them, so maybe we want to look at that in the future.

But for this, I'm fine with it as-is, because I'm convinced enough now that it won't come up for easily predictable or fixable reasons on the user's side.

Reply:
> Could this happen if the user simply doesn't have permission to write to the cache dir?

I'd have to check, but I believe that we already write to a temporary file in the cache directory and the error only occurs trying to swap out the actual file with the temporary one.